src/main/webapp/gast/stat.jsp

```diff
@@ -0,0 +1,170 @@
+        out.println("<nav aria-label=\"Navigation for rows\">");
+        out.println("<ul class=\"pagination\">");
+
+        /*Print Previous Button*/
+        if (currentPage != 1) {
+            out.println(html_prev_button(filterTitle, String.valueOf(recordsPerPage), currentPage,sort,jumpToID));
+        }
+
+        /*Print Pages and highlight current page*/
+        for (int i = 1; i <= nOfPages; i++) {
+            if (currentPage == i) {
+                out.println(html_page_item_current(i));
+            } else {
+                out.println(html_page_item(i, filterTitle, recordsPerPage,sort,jumpToID));
+            }
+        }
+
+        /*Print Next Button*/
+        if (currentPage < nOfPages) {
+            out.println(html_next_button(filterTitle, String.valueOf(recordsPerPage), currentPage,sort,jumpToID));
+        }
+        out.println("</ul>");
+        out.println("</nav>");
+    }
+%>
+<p>
+        
+    <h1>Statistik</h1>
+    <a href="<%=Utils.getBaseUrl(request)%>/gast/quelle">Zurück</a>
+    <br><br>
+    <h3>Liste der Quellen mit Anzahl der Belege
+    </h3>
+    <br>
+    <%
+        print_pagination(out, currentPage, recordsPerPage, filterTitle, nOfPages,sort,jumpToID);
+    %> 
+    <table id="stat1" class="statTable">
+        <thead>        
+            <th>
+                <b>Titel der Quelle</b>
+                <form method="GET">
+                    <input name="filterTitle" type="text" size="40" value="<%=filterTitle %>" placeholder="Titel Filter"/>
+                    <input name="page" type="hidden" value="anzahl_belege"/>
+                    <input name="sort" type="hidden" value="<%=sort %>"/>
+                    <input type="submit" />
+                </form>
+                <%
+                    out.println(html_sort_title_up(currentPage, filterTitle, recordsPerPage,jumpToID,session));
+                    out.println(html_sort_title_down(currentPage, filterTitle, recordsPerPage,jumpToID,session));
+                %>
+            </th>
+            <th>
+                <b>Anzahl Belege</b>
+                <br>
+                <%
+                    out.println(html_sort_belege_up(currentPage, filterTitle, recordsPerPage, jumpToID,session));
+                    out.println(html_sort_belege_down(currentPage, filterTitle, recordsPerPage, jumpToID,session));
+                %>
+            </th>
+        </thead>
+       <tbody>
+            <%
+            for (Quelle q : lst) {
+                out.print("<tr>");
+                out.print("<td width='80%'>");
+                out.print("<a href=\""+Utils.getBaseUrl(request)+"/gast/quelle?ID="+String.valueOf(q.getId())+"\">");
+                out.print(Utils.escapeHTML(q.getBezeichnung()));
+                out.print("</a>");
+                out.print("</td>");
+                out.print("<td width='20%'>");
+                out.print(q.getEinzelbelege().size());
```

Comment:
THIS is the performance problem (not the statement). If you leave the filter empty & sort by belegeDown, then the result list will be rendered. For each Quelle there will be all einzelbelege selected & then counted. Maybe we can provide some kind of shortcut function in the model class that provides better performance (only select the count without returning all Einzelbelege and counting them afterwards?)

Reply:
Yes right.
I have implemented a `count function` in QuelleDB. Now the performance is much better.
(I could not implement the `count function` in the model class, because i needed access to the hibernate session)